### ISSUE TITLE
Message not encrypted

### ISSUE THREAD

**Issue by voznesenskym** [ChatSecure/OTRKit#21]
So - my goal is, in the least number of steps get a message encrypted with libotr on iOS

Currently, with OTRKit, I have a class as an OTR Delegate, and I call 
 [[OTRKit sharedInstance] initiateEncryptionWithUsername:username accountName:accountname protocol:protocol];

and then, a little late on a button press:

[[OTRKit sharedInstance] encodeMessage:@"TEST" tlvs:nil username:username accountName:username protocol:protocol tag:nil];

And when I log the delegate method :

- (void)otrKit:(OTRKit *)otrKit encodedMessage:(NSString *)encodedMessage wasEncrypted:(BOOL)wasEncrypted username:(NSString *)username accountName:(NSString *)accountName protocol:(NSString *)protocol tag:(id)tag error:(NSError *)error {

the encodedMessage does not appear encrypted.

What am I missing?

**Comment by chrisballinger** [comment]
@voznesenskym You'll need bi-directional communication for the key exchange before messages can be encrypted. The injectMessage and encodedMessage delegate methods are not optional, so in those methods you should be ensuring that data reaches the other peer and that on the other end all messages go through decodeMessage. Decoding messages also requires injectMessage to be implemented on the receiving end.

**Comment by voznesenskym** [comment]
Gotcha. So what method do I call to establish a connection to another peer?
In what order does the API get called?

**Comment by chrisballinger** [comment]
OTRKit doesn't rely on any specific network protocol. You can use it with XMPP, or whatever other protocol you prefer.

**Comment by voznesenskym** [comment]
I use http. So how do I ascertain that there was a connection? Where do I
pass in or set details for another client? What is the flow?

I understand that client 1 connects to 2, they handshake, exchange keys,
confirm from ui the connections validity and the conversation is
encrypted... But how does all this happen??

**Comment by voznesenskym** [comment]
I do apologize for basically asking for a 'getting started' guide ,  but even something as simple as you writing out the basic steps of the flow would do wonders in me getting this working...

**Comment by kasas** [comment]
I have the same problem using a XMPP connection through XMPPManager Framework. I can establish the OTR Connection (proved against Adium) the problem is that when I try to encrypt a message it is never done. Do you know why it can happen? I can see both fingerprints :S

**Comment by chrisballinger** [comment]
Did you implement the injectMessage delegate method?

**Comment by kasas** [comment]
Yes All the process to get the session started has been made. But it seems not to encode the messages after the first...

**Comment by kasas** [comment]
Hi, I wanted to ask some help in order to be able to make OTR session in iOS. I have realized that if I start the session in Android or other system the messages are encoded well. The problem is when the session is started in iOS. Do you know any issue in starting sessions??

**Comment by sansuba** [comment]
same problem with me. It is okay with adium and with android how can i solve the problem of encryption with ios to ios. please reply.

**Comment by sansuba** [comment]
everything is okay with android and audium please help to solve iOS to iOS issue

**Comment by chrisballinger** [comment]
I can almost 100% guarantee that you didn't implement injectMessage, or
that incoming messages aren't being passed to decodeMessage

**Comment by sansuba** [comment]
Hi Chris,
I can communicate well. OTR starting and communication is perfect. Thanks for pointing me your guesses. I have implemented them and only thing that I am not decoded ?OTRv2 from receiving side. from your point "that incoming messages aren't being passed to decodeMessage". It works. Thanks.

**Comment by jevonch** [comment]
@sansuba  Have you sucessfully use this OTRKit for implementing yours? Please give me step by step solution..Im newbie in this case

**Comment by chrisballinger** [comment]
Not to be harsh, but if you don't know what you're doing and need step by
step instructions, you probably shouldn't be implementing crypto code quite
yet.

**Comment by sansuba** [comment]
Hi Chris,
can you tell me the process to handle when we lost already encrypted keys from other end. and they still send the same encryption without generating new key pairs. so I am saving OTRFingerprints some where else but do not know to reuse them when app initialize to start OTR session with same user for saved key. 
can you give me some clues to reuse my saved OTRFingerPrints again to decode messages coming from other ends when we do not have InternalFingerprints in Context ?
